dashbuilder/dashbuilder-client/dashbuilder-displayer-client/src/main/java/org/dashbuilder/displayer/client/RendererManager.java

```diff
@@ -106,21 +109,20 @@ public void setDefaultRenderer(DisplayerType displayerType, String rendererName)
     }
 
     public List<RendererLibrary> getRenderersForType(DisplayerType displayerType) {
-        List<RendererLibrary> result = renderersByType.get(displayerType);
-        if (result == null) return new ArrayList<RendererLibrary>();
-        return result;
+        return  renderersByType.getOrDefault(displayerType, new ArrayList<>());
     }
 
     public List<RendererLibrary> getRenderersForType(DisplayerType type, DisplayerSubType subType) {
+        List<RendererLibrary> types  = renderersByType.getOrDefault(type, Collections.emptyList());
+        List<RendererLibrary> subTypes = renderersBySubType.getOrDefault(subType, Collections.emptyList());
         if (type == null) {
```

Comment:
Hello @jhrcek , thanks again for your analysis. I added tests for renderer manager, see commit https://github.com/kiegroup/appformer/pull/573/commits/e38ac054e619096f1bc04dd027b3a0960df71d53Regarding the NPE, I don't see it, perhaps enummap / hashmap does not throw exception.

Regarding the error message, we will add the information to users check if they are not running in offline mode when AF-1751 is implemented. We will add the information if user is running in offline mode or not. 

The idea is that we should cover all types if an user is running in online mode, so in business central they should see it only if they are running in offline mode (because there's only c3 and google there) on dashbuilder same thing, users running in online mode will have Google, which means that all types are covered.

Finally, this exception should never go up for users when all renderer types are implemented by c3!

https://issues.jboss.org/browse/AF-1751

Reply:
You're right, I tried this too and no NPE is thrown when you do `map.get(null)` or `map.getOrDefault(null, defaultVal)` just return null / defaultVal respectively.